api/envoy/config/filter/http/transcoder/v2/transcoder.proto

```diff
@@ -120,4 +120,25 @@ message GrpcJsonTranscoder {
   // not know them beforehand. Otherwise use ``ignored_query_parameters``.
   // Defaults to false.
   bool ignore_unknown_query_parameters = 8;
+
+  // Whether to convert gRPC status headers to JSON.
+  // When trailer indicates a gRPC error and there was no HTTP body, take ``google.rpc.Status``
+  // from the ``grpc-status-details-bin`` header and use it as JSON body.
+  // If there was no such header, make ``google.rpc.Status`` out of the ``grpc-status`` and
+  // ``grpc-message`` headers.
+  // The error details types must be present in the ``proto_descriptor``.
+  //
+  // Example:
+  //
+  // .. code-block:: proto
+  //
+  //     import "google/rpc/error_details.proto";
+  //
+  //     message Dummy {
+  //       google.rpc.RequestInfo request_info = 1;
+  //     }
+  //
+  // If a gRPC service uses the types from the ``google/rpc/error_details.proto``, its proto files
```

Comment:
I'm not grokking the example or this paragraph. Where does `Dummy` appear? In config? On the wire?

Reply:
In the proto generating the descriptor set for transcoding.

Reply:
Docs in this file already have two examples (for the match_incoming_request_route and the ignored_query_parameters options) that show user proto files used for making their proto descriptor.
And this is just another example of what should be written in user proto files.

This `Dummy` type is not used anywhere, but it's required as its contents ensures dependencies on types that have to be in a proto descriptor set.

Reply:
for api approval: @mattklein123 since @htuch is OOO, do you have suggestion here? IMO this is enough given the examples in other fields does similar thing.

Reply:
TBH I don't understand this example either, but I don't have a lot of context. Perhaps you could describe what the output looks like given a particular response?

Reply:
Here is some context:
Upstream server responds with an error in the following headers frame:
```
grpc-status: 5
grpc-status-details-bin: CAUSElJlc291cmNlIG5vdCBmb3VuZA
```
The grpc-status-details-bin header contains base64 encoded protobuf `google.rpc.Status` message. We transcode it into
```
HTTP/1.1 404 Not Found
grpc-status: 5
content-type: application/json
...

{"code":5,"message":"Resource not found"}
```

Now, `google.rpc.Status` has the optional "details" field, which can hold arbitrary user-defined  types. E.g. `grpc-status-details-bin: CAUSBUVycm9yGjYKKXR5cGUuZ29vZ2xlYXBpcy5jb20vaGVsbG93b3JsZC5IZWxsb1JlcGx5EgkKB2RldGFpbHM` will be transcoded into
```
{"code":5,"message":"Error","details":[{"@type":"type.googleapis.com/helloworld.HelloReply","message":"details"}]}
```

Since Envoy knows nothing about this `helloworld.HelloReply` type, it should be in the proto descriptor set provided via config. And the example says "make sure that your details types are in the the proto descriptor set". "If those details types aren't used in any message types of your service, make a dummy message type for them".

Reply:
@ascheglov actually, I don't think you really need the message dummy, only a include of `error_details.proto` or just pass `error_details.proto` to protoc when you generate the descriptor should work. protoc doesn't drop unused message descriptors when it generate them.

Reply:
Can you potentially just update the description/docs to read more like https://github.com/envoyproxy/envoy/pull/8009#discussion_r322780622? That makes sense to me.

Reply:
> @ascheglov actually, I don't think you really need the message dummy, only a include of `error_details.proto` or just pass `error_details.proto` to protoc when you generate the descriptor should work. protoc doesn't drop unused message descriptors when it generate them.

Yes, that's true. With `protoc --include_imports` it's enough to just add an import.

Reply:
@mattklein123 done